### PR TITLE
Various Fixes/Improvements with PanelSyncHandler, Draggables, and Widgets

### PR DESCRIPTION
**Pull request by ghzdude**

### CODE REVIEW

File: src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncHandler.java [hunk 1]
```diff
@@ -107,16 +109,37 @@ public void closePanelInternal() {
...
     @Override
     public void deleteCachedPanel() {
+        if (openedPanel == null || isPanelOpen()) return;
+        boolean canDispose = WidgetTree.foreachChild(openedPanel, iWidget -> {
+            if (!iWidget.isValid()) return false;
+            if (iWidget instanceof ISynced<?>synced && synced.isSynced()) {
+                return !(synced.getSyncHandler() instanceof ItemSlotSH);
+            }
+            return true;
+        }, false);
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncHandler.java:119]: Not a good solution but ill accept it for now

File: src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java [hunk 2]
```diff
@@ -42,9 +42,12 @@ public PanelSyncManager() {}
     public void initialize(String panelName, ModularSyncManager msm) {
         this.modularSyncManager = msm;
         this.panelName = panelName;
-        this.syncHandlers.forEach((mapKey, syncHandler) -> syncHandler.init(mapKey, this));
+        this.syncHandlers.forEach((mapKey, syncHandler) -> {
+            if (!syncHandler.isValid())
+                syncHandler.init(mapKey, this);
+        });
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java:48]: If the sync handler was previously on a different panel sync manager (or still is) this might cause issues

File: src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java [hunk 3]
```diff
@@ -79,8 +82,8 @@ public void onOpen() {
     public void onClose() {
         this.closeListener.forEach(listener -> listener.accept(getPlayer()));
         for (String name : this.subPanels.keySet()) {
-            SyncHandler sh = this.modularSyncManager.getMainPSM().syncHandlers.remove(name);
-            this.modularSyncManager.getMainPSM().reverseSyncHandlers.remove(sh);
+            SyncHandler sh = this.subPanels.get(name).getSyncManager().syncHandlers.remove(name);
+            this.subPanels.get(name).getSyncManager().reverseSyncHandlers.remove(sh);
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java:86]: What is the reason for this change???

File: src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java [hunk 4]
```diff
@@ -194,9 +201,6 @@ public IPanelHandler panel(String key, PanelSyncHandler.IPanelBuilder panelBuild
         if (sh != null) return (IPanelHandler) sh;
         PanelSyncHandler syncHandler = new PanelSyncHandler(panelBuilder, subPanel);
         this.subPanels.put(key, syncHandler);
-        if (isInitialised()) {
-            this.modularSyncManager.getMainPSM().registerPanelSyncHandler(key, syncHandler);
-        }
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java:199]: ???

File: src/main/java/com/cleanroommc/modularui/widget/Widget.java [hunk 5]
```diff
@@ -326,6 +326,7 @@ public W onUpdateListener(Consumer<W> listener, boolean merge) {
     }
 
     public W setEnabledIf(Predicate<W> condition) {
+        setEnabled(condition.test(getThis()));
         return onUpdateListener(w -> setEnabled(condition.test(w)), true);
     }
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/modularui/widget/Widget.java:331]: this might cause issues because the widget is not yet initialised. I already call update function in init which should be enough

**Review by brachy84 — CHANGES_REQUESTED** [review]

File: src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java (hunk 3 above)
Comment (ghzdude) [reply src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java:86]: panel sync handlers are stored in the parent sync manager of the subpanel, which isn't guaranteed to be the main PSM

File: src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java (hunk 4 above)
Comment (ghzdude) [reply src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java:199]: the psm is never initialized at this point, so `isInitialised()` always returns false

Comment (ghzdude) [reply src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java:199]: it also gets registered later when the psm is initialized

File: src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java (hunk 3 above)
Comment (brachy84) [reply src/main/java/com/cleanroommc/modularui/value/sync/PanelSyncManager.java:83]: Why did you change that? I made the psm only be registered to the main psm, because that never gets closed.
Also when you open the 2nd panel on the test tile then close and then open it again an error is thrown

**Review by brachy84 — APPROVED** [review]
